### DISCUSSION TITLE
[Android] How to upload native deubg symbols to playstore?

### DISCUSSION THREAD

**Discussion by rawatnaresh** [fastlane/fastlane#21090] [Q&A]
I'm trying to upload debug symbols to playstore. I've tried 2 different approaches but none of them worked.

## Approach 1st using `mapping_paths`
Initially I tried zipping debug symbols and passed them to `mapping_paths` like this :

```ruby
  desc "Upload app to Staging PlayStore for Internal test"
  lane :deploy_stage_internal do
    build_stage_aab_file
    zip(
      path: "./android/app/build/intermediates/merged_native_libs/stageRelease/out/lib/",
      output_path: "native_debug_symbols.zip",
      verbose: true,
    )
    upload_to_play_store(
      track: "internal",
      skip_upload_images: true,
      skip_upload_screenshots: true,
      release_status: 'draft',
      mapping_paths: ["native_debug_symbols.zip", "./android/app/build/outputs/mapping/stageRelease/mapping.txt"]
    )
  end
```

but it resulted to an error

![Screen Shot 2023-02-20 at 10 57 58](https://user-images.githubusercontent.com/29520305/220014830-d9f31ba3-e572-4164-bc8d-37b4953d9e6f.png)

but zipping debug symbols and uploading them manually works fine. 👍 

### Approach 2nd using `debugSymbolLevel 'FULL' `
According to android [docs](https://developer.android.com/studio/build/shrink-code#android_gradle_plugin_version_41_or_later)

> If your project builds an Android App Bundle, you can automatically include the native debug symbols file in it. To include this file in release builds, add the following to your app's build.gradle file:
>
> `android.buildTypes.release.ndk.debugSymbolLevel = { SYMBOL_TABLE | FULL }`

I updated my app's `build.gradle` file accordingly

```gradle
    buildTypes {
        debug {
            signingConfig signingConfigs.debug
        }
        release {
            // Caution! In production, you need to generate your own keystore file.
            // see https://reactnative.dev/docs/signed-apk-android.
            signingConfig signingConfigs.release
            shrinkResources false
            minifyEnabled enableProguardInReleaseBuilds
            proguardFiles getDefaultProguardFile("proguard-android.txt"), "proguard-rules.pro"

            // https://developer.android.com/studio/build/shrink-code#native-crash-support
            // Setting debugSymbolLevel to 'FULL' automatically include the native debug symbols and mappings file in aab
            // but note that it will increase the file size of your appbundle dramatically.
            ndk {
                debugSymbolLevel 'FULL'
            }

            firebaseCrashlytics {
                nativeSymbolUploadEnabled true
                unstrippedNativeLibsDir file("build/intermediates/merged_native_libs")
            }
        }
    }

```

#### Ran fastlane in Github actions, bundle uploaded successfully with mappings file but no native symbols 😕 
![Screen Shot 2023-02-20 at 11 57 14](https://user-images.githubusercontent.com/29520305/220025067-33d832eb-97c4-4922-bdc3-d7d5ea765182.png)

#### Ran fastlane locally and I can see native debug symbols and mappings attached to the file

![Screen Shot 2023-02-20 at 12 25 24](https://user-images.githubusercontent.com/29520305/220032016-8533d0df-5aeb-44d5-8c74-df96537ac8ce.png)

I'm not sure if I'm missing anything.

##### Command executed

```sh
bundle exec fastlane android deploy_stage_internal --env staging
```

### Environment

<details>
  <pre> 
<details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                         | Value                                                                                 |
| --------------------------- | ------------------------------------------------------------------------------------- |
| OS                          | 12.6.2                                                                                |
| Ruby                        | 2.7.5                                                                                 |
| Bundler?                    | true                                                                                  |
| Git                         | git version 2.37.1 (Apple Git-137.1)                                                  |
| Installation Source         | ~/Documents/xx/xxx/vendor/bundle/ruby/2.7.0/bin/fastlane |
| Host                        | macOS 12.6.2 (21G320)                                                                 |
| Ruby Lib Dir                | ~/.rbenv/versions/2.7.5/lib                                                           |
| OpenSSL Version             | OpenSSL 1.1.1q  5 Jul 2022                                                            |
| Is contained                | false                                                                                 |
| Is homebrew                 | false                                                                                 |
| Is installed via Fabric.app | false                                                                                 |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/                                           |
| Xcode Version               | 14.2                                                                                  |
| Swift Version               | 5.7.2                                                                                 |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
default_platform(:android)

 platform :android do
  ## Readme
  desc "just regenerate fastlane/README.md"
  lane :readme do
  end

  desc 'Bump up version code - to the next highest integer value'
  lane :bumpup_version_code do 
    android_set_version_code(
      gradle_file: "./android/app/build.gradle"
    )
    git_commit( 
      path: "./android/app/build.gradle",
      message: "Android version bump Fastlane"
    )
  end

  desc 'build Staging aab file'
    lane :build_stage_aab_file do
    gradle(
      project_dir: "./android",
      task: "bundle",
      flavor: 'Stage',
      build_type: "Release",
    )
  end

  desc 'build Production aab file'
  lane :build_prod_aab_file do
  gradle(
    project_dir: "./android",
    task: "bundle",
    flavor: 'Prod',
    build_type: "Release",
  )
end

  desc "Upload app to Production PlayStore for Internal test"
  lane :deploy_prod_internal do
    build_prod_aab_file
    upload_to_play_store(
      track: "internal",
      skip_upload_images: true,
      skip_upload_screenshots: true,
      release_status: 'draft',
    )
  end

  desc "Upload app to Staging PlayStore for Internal test"
  lane :deploy_stage_internal do
    gradle(
      project_dir: "./android",
      task: "clean"
    )
    build_stage_aab_file
    upload_to_play_store(
      track: "internal",
      skip_upload_images: true,
      skip_upload_screenshots: true,
      release_status: 'draft',
    )
  end

  error do |lane, exception| 
  end
end
```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
## Android
package_name ENV["ANDROID_PACKAGE_NAME"]
json_key_file "./android/app/key.json"

# For more information about the Appfile, see:
# https://docs.fastlane.tools/advanced/#appfile

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.212.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

| Plugin                             | Version | Update-Status      |
| ---------------------------------- | ------- | ------------------ |
| fastlane-plugin-versioning_android | 0.1.0   | 🚫 Update available |

<details><summary><b>Loaded gems</b></summary>

| Gem                                | Version      |
| ---------------------------------- | ------------ |
| did_you_mean                       | 1.4.0        |
| bundler                            | 2.1.4        |
| uri                                | 0.10.0       |
| rake                               | 13.0.6       |
| CFPropertyList                     | 3.0.3        |
| concurrent-ruby                    | 1.1.10       |
| i18n                               | 1.12.0       |
| minitest                           | 5.16.3       |
| tzinfo                             | 2.0.5        |
| zeitwerk                           | 2.6.6        |
| activesupport                      | 6.1.7        |
| public_suffix                      | 4.0.6        |
| addressable                        | 2.8.0        |
| httpclient                         | 2.8.3        |
| json                               | 2.5.1        |
| algoliasearch                      | 1.27.5       |
| artifactory                        | 3.0.15       |
| atomos                             | 0.1.3        |
| aws-eventstream                    | 1.2.0        |
| aws-partitions                     | 1.714.0      |
| aws-sigv4                          | 1.5.2        |
| jmespath                           | 1.6.2        |
| aws-sdk-core                       | 3.170.0      |
| aws-sdk-kms                        | 1.62.0       |
| aws-sdk-s3                         | 1.119.1      |
| babosa                             | 1.0.4        |
| claide                             | 1.0.3        |
| fuzzy_match                        | 2.0.4        |
| nap                                | 1.1.0        |
| netrc                              | 0.11.0       |
| ffi                                | 1.15.5       |
| ethon                              | 0.16.0       |
| typhoeus                           | 1.4.0        |
| cocoapods-core                     | 1.11.3       |
| cocoapods-deintegrate              | 1.0.5        |
| cocoapods-downloader               | 1.6.3        |
| cocoapods-plugins                  | 1.0.0        |
| cocoapods-search                   | 1.0.1        |
| cocoapods-trunk                    | 1.6.0        |
| cocoapods-try                      | 1.2.0        |
| colored2                           | 3.1.2        |
| escape                             | 0.0.4        |
| fourflusher                        | 2.3.1        |
| gh_inspector                       | 1.1.3        |
| molinillo                          | 0.8.0        |
| ruby-macho                         | 2.5.1        |
| nanaimo                            | 0.3.0        |
| rexml                              | 3.2.5        |
| xcodeproj                          | 1.21.0       |
| cocoapods                          | 1.11.3       |
| colored                            | 1.2          |
| highline                           | 2.0.3        |
| commander                          | 4.6.0        |
| declarative                        | 0.0.20       |
| digest-crc                         | 0.6.4        |
| unf_ext                            | 0.0.8.2      |
| unf                                | 0.1.4        |
| domain_name                        | 0.5.20190701 |
| dotenv                             | 2.8.1        |
| emoji_regex                        | 3.2.3        |
| excon                              | 0.99.0       |
| faraday-em_http                    | 1.0.0        |
| faraday-em_synchrony               | 1.0.0        |
| faraday-excon                      | 1.1.0        |
| faraday-httpclient                 | 1.0.1        |
| multipart-post                     | 2.0.0        |
| faraday-multipart                  | 1.0.4        |
| faraday-net_http                   | 1.0.1        |
| faraday-net_http_persistent        | 1.2.0        |
| faraday-patron                     | 1.0.0        |
| faraday-rack                       | 1.0.0        |
| faraday-retry                      | 1.0.3        |
| ruby2_keywords                     | 0.0.5        |
| faraday                            | 1.10.3       |
| http-cookie                        | 1.0.5        |
| faraday-cookie_jar                 | 0.0.7        |
| faraday_middleware                 | 1.2.0        |
| fastimage                          | 2.2.6        |
| jwt                                | 2.7.0        |
| memoist                            | 0.16.2       |
| multi_json                         | 1.15.0       |
| os                                 | 1.1.4        |
| signet                             | 0.17.0       |
| googleauth                         | 1.3.0        |
| mini_mime                          | 1.1.2        |
| trailblazer-option                 | 0.1.2        |
| uber                               | 0.1.0        |
| representable                      | 3.2.0        |
| retriable                          | 3.1.2        |
| webrick                            | 1.8.1        |
| google-apis-core                   | 0.11.0       |
| google-apis-androidpublisher_v3    | 0.34.0       |
| google-apis-playcustomapp_v1       | 0.12.0       |
| google-apis-iamcredentials_v1      | 0.17.0       |
| google-apis-storage_v1             | 0.19.0       |
| google-cloud-env                   | 1.6.0        |
| google-cloud-errors                | 1.3.0        |
| google-cloud-core                  | 1.6.0        |
| google-cloud-storage               | 1.44.0       |
| mini_magick                        | 4.12.0       |
| naturally                          | 2.2.1        |
| optparse                           | 0.1.1        |
| plist                              | 3.6.0        |
| rubyzip                            | 2.3.2        |
| security                           | 0.1.3        |
| simctl                             | 1.6.10       |
| terminal-notifier                  | 2.0.0        |
| unicode-display_width              | 1.8.0        |
| terminal-table                     | 1.8.0        |
| tty-screen                         | 0.8.1        |
| tty-cursor                         | 0.7.1        |
| tty-spinner                        | 0.9.3        |
| word_wrap                          | 1.0.0        |
| rouge                              | 2.0.7        |
| xcpretty                           | 0.3.0        |
| xcpretty-travis-formatter          | 1.0.1        |
| fastlane-plugin-versioning_android | 0.1.0        |
</details>

*generated on:* **2023-02-20**
</details>

 </pre>
</details>

**Comment by czaefferer** [comment]
For anyone else finding this, the problem with the first approach is, that the zip will contain the "lib" folder, but it should contain the folders below that. See here: https://github.com/fastlane/fastlane/issues/21064

**Comment by dhruvin207** [comment]
Kindly check the provided link: https://github.com/fastlane/fastlane/issues/21064#issuecomment-2036150638.
